tests/test_webhooks.py

```diff
@@ -0,0 +1,355 @@
+
+    def test_Webhook_CreateUser(self):
+        """
+        Test creating a user using webhooks.
+        """
+        # Write temporary config for testing webhooks
+        create_webhook_url = BASE_WEBHOOK_URL + "/webhook_create"
+        data = {
+            "webhooks": [
+                {
+                    "name": "test_webhook",
+                    "action": WEBHOOK_CREATE_USER_ACTION,
+                    "method": "POST",
+                    "url": create_webhook_url,
+                    "payload": {"user_name": "{user_name}", "tmp_url": "{tmp_url}"}
+                },
+                {
+                    "name": "test_webhook_2",
+                    "action": WEBHOOK_CREATE_USER_ACTION,
+                    "method": "POST",
+                    "url": create_webhook_url,
+                    "payload": {"user_name": "{user_name}", "tmp_url": "{tmp_url}"}
+                }
+            ],
+            "providers": "",
+            "permissions": ""
+        }
+
+        with tempfile.NamedTemporaryFile(mode="w") as webhook_tmp_config:
+            yaml.safe_dump(data, webhook_tmp_config, default_flow_style=False)
+
+            # create the magpie app with the test webhook config
+            self.setup_webhook_test(webhook_tmp_config.name)
+
+            utils.get_test_webhook_app(BASE_WEBHOOK_URL)
+
+            utils.TestSetup.create_TestUser(self, override_group_name=get_constant("MAGPIE_ANONYMOUS_GROUP"))
+
+            # Wait for the webhook requests to complete
+            sleep(1)
+
+            # Check if both webhook requests have completed successfully
+            resp = requests.get(BASE_WEBHOOK_URL + "/get_status")
+            assert resp.text == "2"
+
+            # Check if user creation was successful
+            users = utils.TestSetup.get_RegisteredUsersList(self)
+            utils.check_val_is_in(self.test_user_name, users, msg="Test user should exist.")
+
+    def test_Webhook_CreateUser_EmptyUrl(self):
+        """
+        Test creating a user with an empty webhook url.
+        """
+        # Write temporary config for testing webhooks
+        data = {
+            "webhooks": [],
+            "providers": "",
+            "permissions": ""
+        }
+
+        with tempfile.NamedTemporaryFile(mode="w") as webhook_tmp_config:
+            yaml.safe_dump(data, webhook_tmp_config, default_flow_style=False)
+            # create the magpie app with the test webhook config
+            self.setup_webhook_test(webhook_tmp_config.name)
+
+            utils.TestSetup.create_TestUser(self, override_group_name=get_constant("MAGPIE_ANONYMOUS_GROUP"))
+
+            # Wait for the potential webhook requests to complete
+            # In this case, there should be no webhook request to execute
+            sleep(1)
+
+            # Check if user creation was successful even if no webhook were defined in the config
+            users = utils.TestSetup.get_RegisteredUsersList(self)
```

Comment:
Should check that the user has the good status : `UserOKStatus `

Reply:
Good point, I will add this check.